Common/cpp/SharedItems/Shareables.h

```diff
@@ -58,33 +81,40 @@ class Shareable {
 template <typename BaseClass>
 class RetainingShareable : virtual public BaseClass {
  private:
-  std::shared_ptr<JSRuntimeHelper> runtimeHelper_;
-  std::unique_ptr<jsi::Value> remoteValue_;
+  jsi::Runtime *rnRuntime_;
+  jsi::Runtime *uiRuntime_;
```

Comment:
We can't really use a shared pointer for RN runtime simply because we don't have one (we obtain `rnRuntime` as `jsi::Runtime *` casted using `reinterpret_cast` and we usually use it as `jsi::Runtime &`).

We could use a shared pointer for UI runtime but in such case a shareable would affect the lifetime of the runtime which is not what we want.

The current behavior is somewhat similar to `jsi::Value` and `jsi::Runtime` &ndash; you can hold `jsi::Value` longer but they don't extend the lifetime of the `jsi::Runtime`.

Also, this is exactly the same logic that existed before &ndash; previously we would hold `jsRuntimeHelper` which used to hold `jsi::Runtime *rnRuntime;` and `jsi::Runtime *uiRuntime;` and we would call `jsRuntimeHelper->isRNRuntime(rt)` which used to check pointers were equal.

Finally, we can't hold the runtimes here as `jsi::Runtime &` because both fields would have to be initialized in the constructor but we don't know the address of `uiRuntime` until it is called from that runtime.